code/weapon/beam.cpp

```diff
@@ -3294,16 +3289,17 @@ void beam_handle_collisions(beam *b)
 				// hit the ship - again, the innards of this code handle multiplayer cases
 				// maybe vaporize ship.
 				//only apply damage if the collision is not an exit collision.  this prevents twice the damage from being done, although it probably be more realistic since two holes are being punched in the ship instead of one.
+
+
 				if (!b->f_collisions[idx].is_exit_collision)
-					ship_apply_local_damage(&Objects[target], &Objects[b->objnum], &b->f_collisions[idx].cinfo.hit_point_world, beam_get_ship_damage(b, &Objects[target]), b->f_collisions[idx].quadrant);
+					real_damage = beam_get_ship_damage(b, &Objects[target]) * damage_time_mod;
+					ship_apply_local_damage(&Objects[target], &Objects[b->objnum], &b->f_collisions[idx].cinfo.hit_point_world, real_damage, b->f_collisions[idx].quadrant);
 
 				// if this is the first hit on the player ship. whack him
-				if(do_damage)
-				{
-					// Goober5000 - AGH!  BAD BAD BAD BAD BAD BAD BAD BAD bug!  The whack's hit point is in *local*
-					// coordinates, NOT world coordinates!
-					beam_apply_whack(b, &Objects[target], &b->f_collisions[idx].cinfo.hit_point);
-				}
+				// Goober5000 - AGH!  BAD BAD BAD BAD BAD BAD BAD BAD bug!  The whack's hit point is in *local*
+				// coordinates, NOT world coordinates!
+				beam_apply_whack(b, &Objects[target], &b->f_collisions[idx].cinfo.hit_point);
```

Comment:
Does `beam_apply_whack` handle frame time scaling correctly? If it assumes that it is only called every `BEAM_DAMAGE_TIME` milliseconds then this will cause some serious issues at high FPS.